.github/workflows/quality-assurance.yml

```diff
@@ -2,7 +2,11 @@ name: Quality Assurance
 
 on:
   pull_request:
-    branches: [main]
+    branches:
+      - main
+  push:
+    branches:
+      - main
   workflow_dispatch:
```

Comment:
skipped adding inputs to control which job would run, not really required for this PR and could always be done at a later stage